pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -371,10 +372,58 @@ Error: "There are at least two blocks with same name: Build Go project"
 Semaphore assigns its own unique names to nameless `blocks` items, which are
 displayed in the Semaphore 2.0 UI.
 
+### dependencies in blocks
+
+The `dependencies` property of a block is used to define the flow of execution
+between blocks within a pipeline.
+
+To specify that block `C` needs to be run once the blocks `A` and `B` finish,
+the `dependencies` property of the block `C` needs to be populated with corresponding names of the blocks which block `C` depends on.
+In the following example blocks `A` and `B` would be executed in parallel at the
+very beginning of a pipeline.
+
+``` yaml
+version: "v1.0"
+name: Pipeline with dependencies
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: "A"
+    dependencies: []
+    task:
+      jobs:
+      - name: "A"
```

Comment:
Maybe these names should be changed to `Job A` etc. so people would not be confused about which name they should put in `dependencies` list .

Reply:
This is very good notice. I will change it.

Also I consciously decided to name Semaphore entities in a highly generic way, in order not to confuse people with some specific use case who might not be familiar with it. Feedback about this decision is also welcome.

Thanks!